dep_checker/npm_audit.py

```
@@ -39,126 +72,154 @@ class AuditParseError(Exception):
+            return False
+        error = audit_data.get("error")
+        return isinstance(error, dict) and error.get("code") == "ENOLOCK"
+
     def run_npm_install(self, package_dir: Path) -> bool:
-        """Install production dependencies needed for an audit without lifecycle scripts."""
+        """Install non-dev dependencies needed for an audit when no lockfile is present."""
         try:
             logger.info(f"Running npm install in {package_dir}")
-            # Check if npm is available
             npm_check = subprocess.run(
                 ["npm", "--version"],
```

Ruff (0.16.0)

[error] Starting a process with a partial executable path

(S607)